example/thirdparty/ollama-js/build.mill

```diff
@@ -0,0 +1,224 @@
+      def mainFileName = s"fill.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def enableEsm = true
+    }
+
+    object multimodal extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"multimodal.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def enableEsm = true
+    }
+
+    object `pull-progress` extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"pull.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def enableEsm = true
+    }
+
+    object structured_outputs extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def npmDeps = Seq("zod@3.24.1", "zod-to-json-schema@3.24.1")
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"structured-outputs.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def compilerOptions =
+        super.compilerOptions() + (
+          "module" -> ujson.Str("nodenext"),
+          "moduleResolution" -> ujson.Str("nodenext"),
+          "target" -> ujson.Str("es6")
+        )
+    }
+
+    object tools extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"calculator.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def compilerOptions =
+        super.compilerOptions() + (
+          "module" -> ujson.Str("nodenext"),
+          "moduleResolution" -> ujson.Str("nodenext"),
+          "target" -> ujson.Str("es6")
+        )
+    }
+  }
+}
+
+//// The Ollama JavaScript library provides the easiest way to integrate
+//// your JavaScript project with Ollama. https://github.com/ollama/ollama
+//
+//// Project home: https://github.com/ollama/ollama-js
+
+/** Usage
+
+> ./mill compile
+
+> ./mill test
+...
+.../index.test.ts (15 tests) ...
+.../utils.test.ts (5 tests) ...
+...
+...Test Files  2 passed (2)
+...     Tests  20 passed (20)
+...
+
+> ./mill checkFormatEslint # no failures
+
+> ./mill checkFormatPrettier # expected failures
+...Checking formatting...
+[warn] examples/abort/abort-all-requests.ts
+[warn] examples/abort/abort-single-request.ts
+[warn] examples/structured_outputs/structured-outputs-image.ts
+[warn] examples/structured_outputs/structured-outputs.ts
+[warn] examples/tools/calculator.ts
+[warn] examples/tools/flight-tracker.ts
+[warn] src/browser.ts
+[warn] src/constant.ts
+[warn] src/index.ts
+[warn] src/interfaces.ts
+[warn] src/utils.ts
+[warn] test/utils.test.ts
+...
+
+> ./mill examples.abort.run
+...Aborting all requests...
+
+> ./mill examples.fill-in-middle.run
+...
+
+> ./mill examples.multimodal.run
+...
+
+> ./mill examples.pull-progress.run
+...
+
+> ./mill examples.structured_outputs.run
+...
+
+> ./mill examples.tools.run
+...
```

Comment:
Can we add some output assertions on a few more of these `run` commands? That would help us ensure we are actually exercising the code

Reply:
we can run these without installing ollama, I can show the unable to connect to service failures instead?